index.bs

```diff
@@ -1203,6 +1206,74 @@ Gets the {{Error}} object passed to {{SensorErrorEventInit}}.
 </div>
 
 
+<h3 dfn>Find the reporting frequency of a Sensor</h3>
+
+<div algorithm="find the reporting frequency of a sensor">
+
+    : input
+    :: |sensor_instance|, a {{Sensor}} object.
+    : output
+    :: |frequency|, a [=frequency=].
+    1.  Let |frequency| be `null`.
+    1.  Let |f| be |sensor_instance|.{{[[desiredPollingFrequency]]}}.
+        1. if |f| is set,
+            1. set |frequency| to |f| capped by the upper and lower [=current polling frequency=]
+               bounds for the associated [=sensor=].
+        1.  Otherwise,
+            1. user agent can assign |frequency| to an appropriate value.
+    1. return |frequency|.
+</div>
+
+
+<h3 dfn>Report Latest Reading updated</h3>
+
+<div algorithm="report latest reading updated">
+
+    : input
+    :: |sensor_instance|, a {{Sensor}} object.
+    : output
+    :: None
+
+    1.  If |sensor_instance|.{{[[waitingForUpdate]]}} is `true`,
+        1. Abort these steps. 
+    1.  Set |sensor_instance|.{{[[waitingForUpdate]]}} to `true`.
+    1.  Let |lastReportedTimestamp| be the value of |sensor_instance|.{{[[lastEventFiredAt]]}}.
+    1.  If |lastReportedTimestamp| is not set
+        1.  Queue a task to run the [=Notify Sensor Object about new reading=] abstract operation,
+            passing it |sensor_instance| as argument.
+        1.  Abort these steps.
+    1.  Let |reportingFrequency| be result of invoking the [=Find the reporting frequency of a Sensor=]
+        abstract operation.
+    1.  If |reportingFrequency| is `null`
+        1.  Queue a task to run the [=Notify Sensor Object about new reading=] abstract operation,
+            passing it |sensor_instance| as argument.
+        1.  Abort these steps.
+    1.  Let |reportingInterval| be the result of 1 / |reportingFrequency|.
+    1.  Let |timestampDelta| be the result of [=latest reading=]["timestamp"] - |lastReportedTimestamp|.
+    1.  If  |timestampDelta| is greater than or equal to |reportingInterval|
+        1.  Queue a task to run the [=Notify Sensor Object about new reading=] abstract operation,
+            passing it |sensor_instance| as argument.
+        1.  Abort these steps.
+    1.  Let |deferUpdateTime| be the result of |reportingInterval| - |timestampDelta|.
+    1.  User agent must defer queueing a task to run the [=Notify Sensor Object about new reading=] abstract
```

Comment:
Should there be a way to cancel this, e.g., if stop() is called?

[Queuing task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-task) == sensor_task_source.task_queue.add(task);
Do we need to save `'deferred notify'` and then, if stop is called, we would not queue a task?